pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go

```diff
@@ -193,15 +194,15 @@ func (a *index) podWorkloadBuilder(
 	workloadServices krt.Collection[model.ServiceInfo],
 	workloadServicesNamespaceIndex krt.Index[string, model.ServiceInfo],
 	endpointSlices krt.Collection[*discovery.EndpointSlice],
-	endpointSlicesAddressIndex krt.Index[string, *discovery.EndpointSlice],
+	endpointSlicesAddressIndex krt.Index[TargetRef, *discovery.EndpointSlice],
 	namespaces krt.Collection[*v1.Namespace],
 	nodes krt.Collection[*v1.Node],
 ) krt.TransformationSingle[*v1.Pod, model.WorkloadInfo] {
 	return func(ctx krt.HandlerContext, p *v1.Pod) *model.WorkloadInfo {
 		// Pod Is Pending but have a pod IP should be a valid workload, we should build it ,
 		// Such as the pod have initContainer which is initialing.
 		// See https://github.com/istio/istio/issues/48854
-		if (!IsPodRunning(p) && !IsPodPending(p)) || p.Spec.HostNetwork {
+		if kubeutil.CheckPodTerminal(p) {
```

Comment:
Important part is the HostNetwork removal; the other switch is just refactoring to make it more clear

Reply:
This concerns me a _tad_ - we are going from a pod-state denylist to a pod-state allowlist here.

Logically it _should_ be the same, unless K8S adds more phases - notably this won't handle `Unknown` which is AFAIK a valid phase but we probably have other/bigger problems if we hit that.